Add java agent injector manual. (#33)
apache/skywalking-swck: docs/examples/java-agent-injector-usage.md

# Java agent injector Usage

In this example , you will learn how to use the java agent injector in three ways.

## Install injector

> The java agent injector is builded in the operator , so you need to follow [Operator installation instrument](../../README.md#operator) to install the operator firstly.

Reviewer: ```suggestion
The java agent injector is a component of the operator, so you need to follow [Operator installation instrument](../../README.md#operator) to install the operator firstly.
```

## Use default configuration

At first , set the injection label in your namespace as below.

```shell
kubectl label namespace default(your namespace) swck-injection=enabled
```

> Then add `swck-java-agent-injected: "true"` in the labels of yaml file as below.

Reviewer: Why add this label since we actived injector in the namespace

Reviewer 2: > Why add this label since we actived injector in the namespace

I think this is needed because only via namespace annotation you don't know which language agent should be injected, you definitely don't want to inject a Java agent to a Python application Pod

Reviewer: if we have multi-language pods in a namespace, the pod granularity is enough. 

The namespace label is "configuration sugar" instead of a constraint. Thanks for your mentioning, the label name attached to a namespace should be `swck-java-injection`.

Reviewer 2: Yes if that's the case we might want to keep only one annotation name, which can be applied in namespace level or Pod level, instead of 2 different annotation name

Reviewer: make sense to me. @dashanji please use the same annotation to both namespace and pod.

Author: OK , I hava a small question there, the webhook only support namespaceSelector or objectSelector . If either namespace or pod label can active the injector, we need to get every pod's label or the namespace's label in the webhook, the load of the webhook may increase a lot.

Reviewer: After doing some investigation, the injected pods belong to an intersection of these selectors, I thought it's an union:
https://github.com/kubernetes/apiserver/blob/92392ef22153d75b3645b0ae339f89c12767fb52/pkg/admission/plugin/webhook/generic/webhook.go#L143

All right, let's keep it as is.
